src/jit/gentree.cpp

```diff
@@ -15290,9 +15290,8 @@ void Compiler::gtExtractSideEffList(GenTreePtr  expr,
         if (oper == GT_XADD)
         {
             expr->SetOperRaw(GT_LOCKADD);
-#ifndef _TARGET_ARM64_
+            assert(genActualType(expr->gtType) == genActualType(expr->gtOp.gtOp2->gtType));
             expr->gtType = TYP_VOID;
```

Comment:
I think the importer should be changed to set the type to VOID on import of GT_LOCKADD too.  I think this must be missing unless emitTypeSize for TYP_VOID doesn't assert or return 0.

Reply:
I think that, theoretically, it is defined to return the data type. It's in `Compiler::gtExtractSideEffList` that it changes it to `TYP_VOID` when its result isn't used. However, I don't know if it ever actually returns a result.